drivers/storage/portworx/component/disruption_budget.go

```diff
@@ -365,3 +375,122 @@ func RegisterDisruptionBudgetComponent() {
+		userProvidedMinValue = -2
+	}
+
+	// Calculate the minimum number of nodes that should be available
+	quorumValue := int(math.Floor(float64(storageNodesCount)/2) + 1)
+	calculatedMinAvailable := quorumValue
+	// When non disruptive upgrades is disabled
+	if cluster.Annotations != nil && cluster.Annotations[pxutil.AnnotationsDisableNonDisruptiveUpgrade] == "true" {
+		if cluster.Annotations[pxutil.AnnotationStoragePodDisruptionBudget] == "" {
+			calculatedMinAvailable = storageNodesCount - 1
+		} else if (userProvidedMinValue < quorumValue || userProvidedMinValue >= storageNodesCount) && userProvidedMinValue != c.annotatedMinAvailable {
+			calculatedMinAvailable = storageNodesCount - 1
+			errmsg := fmt.Sprintf("Invalid minAvailable annotation value for storage pod disruption budget. Using default value: %d", calculatedMinAvailable)
+			c.recorder.Event(cluster, v1.EventTypeWarning, util.InvalidMinAvailable, errmsg)
+		} else if userProvidedMinValue >= quorumValue && userProvidedMinValue < storageNodesCount {
+			calculatedMinAvailable = userProvidedMinValue
+		}
+	} else {
+		// When non disruptive upgrades is enabled
+		if userProvidedMinValue >= storageNodesCount && userProvidedMinValue != c.annotatedMinAvailable {
+			errmsg := fmt.Sprintf("Invalid minAvailable annotation value for storage pod disruption budget. Using default value: %d", calculatedMinAvailable)
+			c.recorder.Event(cluster, v1.EventTypeWarning, util.InvalidMinAvailable, errmsg)
+		} else if userProvidedMinValue >= quorumValue && userProvidedMinValue < storageNodesCount {
+			calculatedMinAvailable = userProvidedMinValue
+		}
+	}
+	c.annotatedMinAvailable = userProvidedMinValue
+	errors := []error{}
+	downNodesCount := numPxNodesDown
+	// Update minAvailable to 0 for nodes in nodesToUpgrade
+	for _, node := range nodesToUpgrade {
+		// Ensure that portworx quorum or valid minAvailable provided by user is always maintained
+		if downNodesCount >= storageNodesCount-calculatedMinAvailable {
+			break
+		}
+		k8snode := cordonedPxNodesMap[node]
+		PDBName := "px-" + k8snode
+		minAvailable := intstr.FromInt(0)
+		pdb := &policyv1.PodDisruptionBudget{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            PDBName,
+				Namespace:       cluster.Namespace,
+				OwnerReferences: []metav1.OwnerReference{*ownerRef},
+			},
+			Spec: policyv1.PodDisruptionBudgetSpec{
+				MinAvailable: &minAvailable,
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						constants.LabelKeyClusterName:      cluster.Name,
+						constants.OperatorLabelNodeNameKey: node,
+					},
+				},
+			},
+		}
+		err = k8sutil.CreateOrUpdatePodDisruptionBudget(c.k8sClient, pdb, ownerRef)
+		if err != nil {
+			logrus.Warnf("Failed to update PDB for node %s: %v", node, err)
+			errors = append(errors, err)
+		} else {
+			downNodesCount++
+		}
+	}
+	return utilerrors.NewAggregate(errors)
```

Comment:
Then should we also return nil for cluster wide pdb?

Reply:
It is okay to have it return on create especially with the cluster wide because that is the only PDB that is getting created. 

Here the updates will happen frequently and automatically without user intervention. So let's just keep trying in the background with log messages. Returning error will raise an event and the user will unnecessarily be worried.